test/end2end_test.dart

```diff
@@ -18,56 +18,47 @@ final _goldenDir = p.join('test', 'goldens', 'end2end');
 final _pubDevUri = Uri.parse('https://pub.dartlang.org/');
 
 void main() {
-  late String tempDir;
-  late PackageAnalyzer analyzer;
-  http.Client? httpClient;
-  late HttpServer httpServer;
-
-  setUpAll(() async {
-    tempDir = Directory.systemTemp
-        .createTempSync('pana-test')
-        .resolveSymbolicLinksSync();
-    final pubCacheDir = p.join(tempDir, 'pub-cache');
-    final panaCacheDir = p.join(tempDir, 'pana-cache');
-    final dartConfigDir = p.join(tempDir, 'config', 'dart');
-    final flutterConfigDir = p.join(tempDir, 'config', 'flutter');
-    final goldenDirLastModified = await _detectGoldenLastModified();
-    Directory(pubCacheDir).createSync();
-    Directory(panaCacheDir).createSync();
-    Directory(dartConfigDir).createSync(recursive: true);
-    Directory(flutterConfigDir).createSync(recursive: true);
-    httpClient = http.Client();
-    httpServer = await _startLocalProxy(
-      httpClient: httpClient,
-      blockPublishAfter: goldenDirLastModified,
-    );
-    analyzer = PackageAnalyzer(await ToolEnvironment.create(
-      dartSdkConfig: SdkConfig(configHomePath: dartConfigDir),
-      flutterSdkConfig: SdkConfig(configHomePath: flutterConfigDir),
-      pubCacheDir: pubCacheDir,
-      panaCacheDir: panaCacheDir,
-      pubHostedUrl: 'http://127.0.0.1:${httpServer.port}',
-      dartdocVersion: 'any',
-    ));
-  });
-
-  tearDownAll(() async {
-    await httpServer.close(force: true);
-    httpClient!.close();
-    Directory(tempDir).deleteSync(recursive: true);
-  });
-
   void verifyPackage(
     String package,
     String version, {
     bool skipDartdoc = false,
   }) {
     final filename = '$package-$version.json';
     group('end2end: $package $version', () {
+      late String tempDir;
+      late PackageAnalyzer analyzer;
+      http.Client? httpClient;
+      late HttpServer httpServer;
       late final Map<String, Object?> actualMap;
 
       setUpAll(() async {
```

Comment:
Should all this setup code not be in a `setUp` rather than `setUpAll`?

Reply:
We have multiple `test()` methods, I'd rather keep it like that.

Reply:
But doesn't that still create only one analysis context for all tests?

Reply:
`setupAll` runs only a single time for all the tests in the group, I think that's what we want here. Not sure: what do you see as a problem here?

Reply:
Ah - I think I saw it as residing at a higher level - sorry.